packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -132,6 +153,7 @@ function DocSidebarItemCategory({
           <DocSidebarItem
             tabIndex={collapsed ? '-1' : '0'}
             key={childItem.label}
+            link={link}
```

Comment:
Why bind this to the children elements? Only the category with `link` should have a link associated right?